pkg/agent/validations.go

```diff
@@ -63,6 +62,129 @@ type validationTrace struct {
 
 var previousValidations []validationTrace
 
+type validationResultHistory struct {
+	numFailures     map[string]int
+	seen            map[string]bool
+	currentStatus   map[string]string
+	currentMessage  map[string]string
+	previousStatus  map[string]string
+	previousMessage map[string]string
+}
+
+func checkValidations(cluster *models.Cluster, validationHistory map[string]*validationResultHistory, log *logrus.Logger) error {
+	if validationHistory["cluster"] == nil {
```

Comment:
nit: for readability, I'd prefer the `comma ok` idiom, ie `if _, ok := validationHistory["cluster"]; !ok {` for checking the existence of the element in the map